fairscale/experimental/nn/distributed_pipeline/partition_handler.py

```diff
@@ -275,14 +275,19 @@ def run_pipeline(self, pipeline_record_rref: rpc.RRef) -> Optional[Tensor]:
         pipeline_record = pipeline_record_rref.local_value()
         self.run(pipeline_record)
 
+        result: Optional[Tensor] = None
+
         if not pipeline_record.consumers:
-            result = microbatch.gather(pipeline_record.batches)
-            assert len(result) == 1
-            result = result[0]
+            gather_result = microbatch.gather(pipeline_record.batches)
+            assert len(gather_result) == 1
+            result = gather_result[0]
             s0 = current_stream(result.device)
             if is_cuda(s0):
                 # TODO. Investigate why this is needed and remove it if possible.
                 as_cuda(s0).synchronize()
-            return result
 
-        return None
+        # TODO: There seems to be a memory leak that is solved by following line.
```

Comment:
Did you confirm the memory footprint reduction by profiling?

Reply:
Before the change nvidia-smi was showing continues increase in memory consumption, and after the fix memory size on gpu was stabilized.